src/stdune/io.ml

```diff
@@ -109,18 +109,43 @@ struct
           };
         f lb)
 
-  (* This function is a copy&paste from Jane Street's [Stdio.In_channel.input_all]. *)
-  let read_all t =
+  let read_all =
     (* We use 65536 because that is the size of OCaml's IO buffers. *)
     let chunk_size = 65536 in
-    let buffer = Buffer.create chunk_size in
-    let rec loop () =
-      Buffer.add_channel buffer t chunk_size;
-      loop ()
+    (* Generic function for channels such that seeking is unsupported or broken *)
+    let read_all_generic t buffer =
+      try
+        while true do
+          Buffer.add_channel buffer t chunk_size
+        done;
+        assert false
```

Comment:
Why `assert false` instead of the old implementation (which didn't need an `assert false`)?

Reply:
I didn't realise that we needed the `assert false` initially and thought the infinite while loop was simpler. But since we need the `assert false`, agreed that the old implementation was better. I changed it back.